### PR TITLE
Boos 215/feature/email service

### PR DESCRIPTION
**Pull request by koutsilisk**
Im this pr we introduce:
- Email service class implementation with simple functions to create and send emails

### CODE REVIEW

File: backend/config.py [hunk 1]
```diff
@@ -76,3 +79,8 @@ class TestConfig(BaseConfig):
 
     # JWT CONFIGURATION
     JWT_ACCESS_TOKEN_EXPIRES = 60  # 1 minute
+
+    # Email configuration
+    MAIL_BACKEND = (
+        "locmem"  # REF https://www.waynerv.com/flask-mailman/#in-memory-backend
```
Comment (Chichilele) [inline backend/config.py:85]: excellent! Maybe we can use this as default and smtp only for prod?

File: backend/project/auth/services/email.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
+from flask_mailman import EmailMessage
+from flask_mailman.message import MIMEBase
+from flask_mailman.backends.smtp import EmailBackend
+from typing import List, Union, Dict, Tuple
+
+
+class Email_service:
+    """Service class from email related functionality"""
+
+    def __init__(self, connection: EmailBackend) -> None:
```
Comment (Chichilele) [inline backend/project/auth/services/email.py:10]: document constructor's parameter

File: backend/project/auth/services/email.py [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+        self.connection = connection
+
+    def create(
```
Comment (Chichilele) [inline backend/project/auth/services/email.py:13]: Might be worth creating a `send_confirmation_email` function for encapsualtion

**Review by Chichilele — APPROVED** [review]
Looks good and seems to work. Are you not doing unit tests though? The service test don't mock anything

File: backend/project/auth/services/email.py (hunk 3 above)
Comment (koutsilisk) [reply backend/project/auth/services/email.py:13]: I thought this is another ticket, for this ticket it's the service impl. Ticket :https://intellium-ai.atlassian.net/jira/software/projects/BOOS/boards/1?selectedIssue=BOOS-208

Comment (Chichilele) [reply backend/project/auth/services/email.py:13]: Yeah so for this ticket I'd probably add a method to the service, loading an HTML template or something like that

**Comment by koutsilisk** [comment]
> 
When testing flag is True in cofing, it uses the MAIL_BACKEND = 'locmem' ( also explicitly added it to the config file ). The flask_mailman documentation states that this is used for testing purposes. It does not send actual emails with that so i thought it was fine to not mock the flask_mailman functions since we don't add any logic, we just port.

File: backend/config.py (hunk 1 above)
Comment (koutsilisk) [reply backend/config.py:85]: i think the "console" one is better as default ( prints emails at stdout, not sending emails )

**Review by Chichilele — APPROVED** [review]

**Pull request merged**